download-response-data.js

```diff
@@ -0,0 +1,115 @@
+/**
+ * Connect to Firestore using a service account, and download participant response data.
+ *
+ * Usage:
+ *   npm run firebase:download -- studyId participantId [sessionNumber] [outputRoot]
+ *
+ *   studyId        must be one of the study Ids that a participant can log in with
+ *   participantId  must be one of the participant Ids that a participant can log in with
+ *   sessionNumber  optional number to select which session to download, defaults to "latest", ie the most recent session
+ *   outputRoot     optional path to the folder where data should be saved, defaults to ".", ie the current folder
+ *
+ * Before using this, you have to do a little setup.
+ *
+ * You must be a project owner for the Firebase project that you want to get data from.
+ * Then you must set up a service account to allow your local machine to access the data.
+ * The official docs for this are here: https://cloud.google.com/docs/authentication/production
+ *
+ * And here's a summary:
+ *  - Visit the Firebase console at https://console.firebase.google.com/
+ *  - Click on the Firebase project you want to get data from
+ *  - In the top left, click on the gear next to "Project Overview"
+ *  - Click on "Users and permissions"
+ *  - Click on the "Service accounts" tab
+ *  - Near the bottom click "Generate new Private key"
+ *  - Save the key on your local machine and remember where you put it
+ *
+ * This Private key is a secret, so be careful with it!
+ * Do not share it.
+ * You may wish to restrict the file permissions on it so only you can access it.
+ * For example: "chmod 600 path/to/serviceAccountKey.json".
+ * You may also wish to delete it when you're done.
+ * Do not share it.
```

Comment:
Maybe we should also add in that people shouldn't commit it into the repo too

Reply:
Sounds like a good idea, I'll add that.